xarray/backends/rasterio_.py

```diff
@@ -120,7 +143,36 @@ def open_rasterio(filename):
     # Maybe we'd like to parse other attributes here (for later)
 
     data = indexing.LazilyIndexedArray(RasterioArrayWrapper(riods))
+
+    # this lets you write arrays loaded with rasterio
+    data = indexing.CopyOnWriteArray(data)
+    if cache and (chunks is None):
+        data = indexing.MemoryCachedArray(data)
+
     result = DataArray(data=data, dims=('band', 'y', 'x'),
                        coords=coords, attrs=attrs)
+
+    if chunks is not None:
+        # Logic borrowed from open_dataset
+        try:
+            from dask.base import tokenize
+        except ImportError:
+            # raise the usual error if dask is entirely missing
+            import dask
+            if LooseVersion(dask.__version__) < LooseVersion('0.6'):
+                raise ImportError(
+                    'xarray requires dask version 0.6 or newer')
```

Comment:
dask v0.6 is pretty old now, maybe better to replace this whole conditional logic with:
```
import dask
from dask.base import tokenize
```